manager/dispatcher/orphans.go

```diff
@@ -0,0 +1,133 @@
+package dispatcher
+
+import (
+	"sync"
+	"time"
+
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/log"
+	"github.com/docker/swarmkit/manager/state/store"
+	"golang.org/x/net/context"
+)
+
+// downNodeStore defines a type to store the nodes which are currently
+// down.
+type downNodeStore struct {
+	// downNodeCheckInterval specifies the frequency at which the
+	// down nodes are scanned.
+	downNodeCheckInterval time.Duration
+
+	// downNodeOrphanThreshold specifies the how long before the
+	// tasks that belong to down nodes are moved to ORPHANED
+	// state.
+	downNodeOrphanThreshold time.Duration
+
+	// nodes is a map of nodes with nodeID as the key and scan
+	// count which tracks the timer expiry as the value.
+	nodes map[string]int
```

Comment:
Wondering if maybe it would make sense to put this on the actual `Node` objects so it survives leader failover. It's already structured in a way that keeps write volume low.